### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by krs013**
Here are a few bugfixes, including that font fix I promised, fixes for #170 and #187, and fixes for a couple runtime bugs I encountered on the way.

I also did add the script I used to make the font along with a makefile to guide the process. It should work on Mac/Linux but it only needs to be run to generate the file that's already committed, so it's mostly there if we want to change it or retrace our steps. It's pretty sketchy code, though, please don't judge, haha.

I don't know if there's a sure way for me to test the font/pip situation, but `pip install .` made an installation in a venv that could use the font properly.

There might be some changes needed if the tests fail. I couldn't get them all to pass locally, but I don't think that's always an indicator of how they'll do on Travis.

In all, these should all be pretty simple and small tweaks. Feel free to change anything that doesn't seem right.

### CODE REVIEW

File: pyboy/core/cartridge/cartridge.py [hunk 1]
```diff
@@ -57,7 +57,15 @@ def load_romfile(filename):
...
 
     logger.debug(f"Loading ROM file: {len(romdata)} bytes")
+    if len(romdata) == 0:
+        logger.error("ROM file is empty!")
+        raise Exception("Empty ROM file")
+
     banksize = 16 * 1024
+    if len(romdata) % banksize != 0:
+        logger.error("Unexpected ROM file length")
+        raise Exception("Bad ROM file size")
```
Comment (Baekalfen) [inline pyboy/core/cartridge/cartridge.py:67]: Exceptions are not working so well in Cython, but this isn't a central part of any important logic, so let's just leave it like this.

File: pyboy/pyboy.py [hunk 2]
```diff
@@ -224,12 +227,16 @@ def stop(self, save=True):
             save (bool): Specify whether to save the game upon stopping. It will always be saved in a file next to the
                 provided game-ROM.
         """
-        if not self.stopped:
+        # If __init__ errors, __del__ will call try to call stop() but
+        # self.stopped doesn't exist yet
+        if hasattr(self, "stopped") and not self.stopped:
```
Comment (Baekalfen) [inline pyboy/pyboy.py:232]: Does `hasattr` work in Cython?

**Review by Baekalfen — APPROVED** [review]
Great job! Not a single thing I would have done differently

**Comment by Baekalfen** [comment]
Your history is so neat, that I'll rebase it ;)

**Comment by Baekalfen** [comment]
Great, then it merged anyway...

File: pyboy/core/cartridge/cartridge.py (hunk 1 above)
Comment (krs013) [reply pyboy/core/cartridge/cartridge.py:67]: They seemed okay when I tested them, but if it becomes a problem some day we can deal with it. It's not likely that these will fire for anyone--but I made a mistake once that caused a rom to have 0 length, and this is easier to understand than a segfault, haha.

File: pyboy/pyboy.py (hunk 2 above)
Comment (krs013) [reply pyboy/pyboy.py:232]: It seems to? I'm not sure *how* it works, but before I added this there were some weird errors happening on __del__. Maybe `hasattr` returns false if the property isn't initialized or something.

Comment (krs013) [reply pyboy/pyboy.py:232]: Update: as we found later, it definitely doesn't work. I pushed a couple more commits that fix it, but I think I'll have to open a separate PR for that. Sorry!

**Pull request merged**